src/Polly.Core/Utils/PipelineComponent.cs

```diff
@@ -11,7 +11,7 @@ namespace Polly.Utils;
 /// <remarks>
 /// The component of the pipeline can be either a strategy, a generic strategy or a whole pipeline.
 /// </remarks>
-internal abstract partial class PipelineComponent
+internal abstract partial class PipelineComponent : IDisposable, IAsyncDisposable
```

Comment:
Shouldn't we implement the disposable pattern here and have the derived classes override a protected method instead?

Reply:
I wouldn't  complicate it here too much. It's an internal API and we know that the disposal by registry is under our control.